server/admin/api.go

```diff
@@ -180,3 +184,43 @@ func (s *Server) apiAccountInfo(w http.ResponseWriter, r *http.Request) {
 	}
 	writeJSON(w, acctInfo)
 }
+
+// apiBan is the handler for the '/account/{accountID}/ban?rule=RULE' API request.
+func (s *Server) apiBan(w http.ResponseWriter, r *http.Request) {
+	acctIDStr := chi.URLParam(r, accountIDKey)
+	acctIDSlice, err := hex.DecodeString(acctIDStr)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("could not decode accout id: %v", err), http.StatusBadRequest)
+		return
+	}
+	if len(acctIDSlice) != account.HashSize {
+		http.Error(w, "account id has incorrect length", http.StatusBadRequest)
+		return
+	}
+	ruleStr := r.URL.Query().Get(ruleToken)
+	if ruleStr == "" {
+		http.Error(w, "rule not specified", http.StatusBadRequest)
+		return
+	}
+	ruleInt, err := strconv.Atoi(ruleStr)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("bad rule: %v", err), http.StatusBadRequest)
+		return
+	}
+	if ruleInt < 1 || ruleInt >= int(account.MaxRule) {
+		http.Error(w, "bad rule: not known or not punishable", http.StatusBadRequest)
```

Comment:
Was about to push but the no golangci-lint complains about these two unneeded Sprintf calls (gosimple).  Fixed.